work in progress, but basic read functionality is there

ingmargoudt opened the pull request.

ingmargoudt committed:
first commit of forum changes
creating client web\js\services
creating server entities

ingmargoudt committed:
hiring Winston for file-logging

ingmargoudt committed:
forum board are now working. data gets returned from the database and shown on the view

ingmargoudt committed:
database change!!!!! moved the forum_posts.title to forum_topics.title
run createtopictilecolumn.sql
run createtopictitle.sql

ingmargoudt committed:
moving db calls to the entity from the router
adding winston calls (WIP)

ingmargoudt committed:
topics can be accessed

bsparks commented on src/client/web/js/services/category.js:
```diff
@@ -0,0 +1,28 @@
+
+                return cats;
+            }, function(err) {
+                $log.error('Error retreiving categories from server.', err);
```
need to inject $q and return $q.reject(); or else the promise will die here

bsparks commented on src/client/web/js/services/forum.js:
```diff
@@ -0,0 +1,44 @@
+            });
+    };
+
+    Forum.getLatestOnlineUsers = function(){
```
seems like this method wouldn't really be forum specific, maybe in User service?

bsparks commented on src/client/web/js/services/forum.js:
```diff
@@ -0,0 +1,44 @@
+    Forum.getLatestOnlineUsers = function(){
+
+        var deferred = $q.defer();
+        $http.get('/api/onlineusers' )
```
this is already a promise/deferred, you can just return it, no need for the extra deferred variable

bsparks commented on src/client/web/js/services/forum.js:
```diff
@@ -0,0 +1,44 @@
+                deferred.resolve(response);
+            }, function(err) {
+                $log.error('Error getting board from server', err);
+                deferred.reject(err);
```
if you change to returning the http promise directly, here you return $q.reject(err);

bsparks commented on src/client/web/js/services/post.js:
```diff
@@ -0,0 +1,26 @@
+        angular.copy(json || {}, this);
+    };
+
+    Post.prototype.$save = function(boardId, topicId) {
```
seems like boardId and topicId would be properties of the post, so you wouldn't have to pass them in?

bsparks commented on src/client/web/js/services/topic.js:
```diff
@@ -0,0 +1,46 @@
+    };
+
+    Topic.getTopic = function(topidId){
+        var deferred = $q.defer();
```
same with these, don't need the extra deferreds

bsparks commented on src/client/web/views/board.html:
```diff
@@ -3,16 +3,26 @@
-                    <a class="thread-name" href="/forum/{{board.id}}/topics/{{ post.topic_id }}">{{ post.title }}</a>
+           
+               
+                <table>
```
didn't like using a more HTML5 approach than tables?

bsparks commented on src/server/entity/board.js:
```diff
@@ -0,0 +1,132 @@
+
+            if(results.length === 0) {
+                log('no topics found');
+                deferred.resolve([]);
```
prolly want to return here or just leave it out since below it'll still resolve with topics = []

bsparks commented on src/server/entity/category.js:
```diff
@@ -0,0 +1,50 @@
+    var Q = require('q'),
+        _ = require('underscore');
+
+    var Category = Class.extend({
```
prolly should name this guy ForumCategory since we might have other types of Categories

bsparks commented on src/server/entity/category.js:
```diff
@@ -0,0 +1,50 @@
+        }
+    });
+
+    Category.get = function(bookId) {
```
copy/pasted "bookId" :)

bsparks commented on src/server/entity/forum.js:
```diff
@@ -0,0 +1,60 @@
+        }
+    });
+
+    Forum.getOnlineUsersLastDay = function() {
```
much like in the client, should prolly move this to user service

bsparks commented on src/server/entity/forum.js:
```diff
@@ -0,0 +1,60 @@
+
+    Forum.getOnlineUsersLastDay = function() {
+        var deferred = Q.defer();
+        db.query('SELECT name from bcs_users WHERE last_session > '+  (Date.now()/1000 - 86400) + ' ORDER BY last_session', function(err,results) {
```
also note that I don't think I ever added anything for last_session in the "new" server code

ingmargoudt replied:
Thats legacy from the current forum, copied it over.

bsparks commented on install/createtopictilecolumn.sql:
```diff
@@ -0,0 +1,41 @@
+/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;
+CREATE DATABASE /*!32312 IF NOT EXISTS*/`ironbane` /*!40100 DEFAULT CHARACTER SET latin1 */;
+
+USE `ironbane`;
```
this will make it harder to execute on ironbane_dev

ingmargoudt committed:
Merge branch 'master' of http://github.com/ironbane/IronbaneServer

ingmargoudt committed:
- added some extra svg used by the forum (for now)
- corrected Ben's review comments (thanks!)
- created a directive: userlink
- start of user profile

ingmargoudt committed:
adding yuidoc

ingmargoudt committed:
getting that forum to work!
